common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -181,10 +200,73 @@ def student_view(self, context):
                   'ajax_url': self.system.ajax_url,
                   }
 
-        fragment.add_content(self.system.render_template('seq_module.html', params))
+        fragment.add_content(self.system.render_template("seq_module.html", params))
 
         return fragment
 
+    def _time_limited_student_view(self, context):
+        """
+        Delegated rendering of a student view when in a time
+        limited view. This ultimately calls down into edx_proctoring
+        pip installed djangoapp
+        """
+
+        # None = no overridden view rendering
+        view_html = None
+
+        # Is this sequent part of a timed or proctored exam?
+        proctoring_service = self.runtime.service(self, 'proctoring')
+        credit_service = self.runtime.service(self, 'credit')
+
+        # Is the feature turned on and do we have all required services
+        # Also, the ENABLE_PROCTORED_EXAMS feature flag must be set to
+        # True and the Sequence in question, should have the
+        # fields set to indicate this is a timed/proctored exam
+        feature_enabled = (
+            proctoring_service and
+            credit_service and
+            proctoring_service.is_feature_enabled()
+        )
+        if feature_enabled:
+            user_id = self.runtime.user_id
+            user_role_in_course = 'staff' if self.runtime.user_is_staff else 'student'
```

Comment:
Is this properly handling CourseStaff (not global staff) as well as user.is_staff (global staff)? I'm not familiar with this flag on the runtime.

Often we call has_access(request.user, 'staff', course, course.id), as that handles both CourseStaff, and global staff.